privacyidea/lib/token.py

```diff
@@ -191,23 +191,29 @@ def _create_token_query(tokentype=None, realm=None, assigned=None, user=None,
                                           TokenRealm.token_id ==
                                           Token.id)).distinct()
 
-    if resolver is not None and resolver.strip("*"):
+    stripped_resolver = None if resolver is None else resolver.strip("*")
+    stripped_userid = None if userid is None else userid.strip("*")
+    if stripped_userid or stripped_resolver:
+        # Join the search with the token owner
+        sql_query = sql_query.filter(TokenOwner.token_id == Token.id)
```

Comment:
This will result in duplicate comparisons in the WHERE clauses if we, for example, pass a ``resolver`` *and* ``assigned=True``:
```python
>>> get_tokens(resolver='passwd', assigned=True)
```
emits
```sql
SELECT ...
FROM token, tokenowner 
WHERE token.id = tokenowner.token_id AND tokenowner.token_id = token.id AND tokenowner.resolver = ?
```
... but even though it looks weird, it works, so I guess this is OK for now.